### PR TITLE
Slack app intercept

### PR DESCRIPTION
**Pull request by haroon-ali-dev**
### Summary
A Slack app that enforces users to complete a required profile field when joining the workspace and sends reminders if their field is blank when they post in channels that the app has been added to.

### Changes Made
1. **User Join Workflow:**
   - Added functionality to send a welcome message to users joining the workspace, prompting them to set a specific profile field.
   
2. **Message Post Workflow:**
   - Implemented a check for the specified profile field when a user posts a message in a channel where the app is active.
   - If the field is blank, a reminder message is sent to the user.

3. **Testing:**
   - Tests using `Jest` and `MSW`.

### Installation instructions
Run `npm install`.

### Testing Instructions
Run `node --experimental-vm-modules node_modules/jest/bin/jest.js`.

### CODE REVIEW

File: netlify/functions/slack-app.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                // Check if the 'Is/Was Trainee' field is empty
+                if (!userProfile.profile.fields['Xf067WCKHDB6']) {
+                    // Send a message to the user about the missing profile field
+                    await client.chat.postMessage({
+                        channel: message.user,
+                        text: 'Please make sure to set your "Is/Was Trainee" profile field.',
+                    });
+                }
+            } catch (error) {
+                console.error('Error checking user profile:', error);
```
Comment (EneVladMihai) [inline netlify/functions/slack-app.js]: There's also the postMessage which can throw an error.

File: netlify/functions/slack-app.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        // Check if the message is from a user in a channel (not a bot message)
+        if (message.subtype !== 'bot_message' && message.user) {
+            try {
+                // Retrieve user profile information
+                const userProfile = await client.users.profile.get({
+                    user: message.user,
+                });
+
+                // Check if the 'Is/Was Trainee' field is empty
+                if (!userProfile.profile.fields['Xf067WCKHDB6']) {
```
Comment (EneVladMihai) [inline netlify/functions/slack-app.js]: Should the field ID be part of the environment variables?

File: requests/message.json
Comment (EneVladMihai) [inline requests/message.json:1]: Feels like requests is not suggestive enough for what's in the folder. How about `sample_data`?

File: netlify/functions/slack-app.js (hunk 1 above)
Comment (EneVladMihai) [reply netlify/functions/slack-app.js]: On a different note, does netlify have any guidance on error reporting from lambda functions?

Comment (haroon-ali-dev) [reply netlify/functions/slack-app.js]: I've updated the error messages.
Netlify's guides say that all console logs can be accessed in the serverless function logs:
https://docs.netlify.com/functions/logs/

So I think this implementation is alright maybe?

File: netlify/functions/slack-app.js (hunk 2 above)
Comment (haroon-ali-dev) [reply netlify/functions/slack-app.js]: Switched to env variables.

File: requests/message.json
Comment (haroon-ali-dev) [reply requests/message.json:1]: Changed folder name.

**Review by EneVladMihai — APPROVED** [review]

**Review by EneVladMihai — APPROVED** [review]

**Pull request merged**